commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -28,13 +28,14 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
       CartExtension.CartDeliveryGroupList cartDeliveryGroups = cart.getCartDeliveryGroups();
       CartExtension.CartDeliveryGroup cartDeliveryGroup = cartDeliveryGroups.get(0);
 
-      // Map cart ID to cart item.
+      // Map cart ID to cart item with type Product.
       CartExtension.CartItemList cartItemCollection = cart.getCartItems();
       Map<String, CartExtension.CartItem> cartItemById = new Map<String, CartExtension.CartItem>();
       for (Integer i = (cartItemCollection.size() - 1); i >= 0; i--) {
-        cartItemById.put(cartItemCollection.get(i).getId(), cartItemCollection.get(i));
+        if (cartItemCollection.get(i).getType() == CartExtension.SalesItemTypeEnum.PRODUCT) {
```

Comment:
just add comment at the top that cart item list will have products and shipping cart item and the explanation and client can customized it as per their need.